builder/main.py

```diff
@@ -147,10 +277,27 @@ def generate_uf2(target, source, env):
 upload_actions = []
 upload_source = target_firm
 
+def UploadUF2ToDisk(target, source, env):
+    assert "UPLOAD_PORT" in env
+    progname = env.subst("$PROGNAME")
+    ext = "uf2"
+    fpath = join(env.subst("$BUILD_DIR"), "%s.%s" % (progname, ext))
+    if not isfile(fpath):
+        print(
+            "Firmware file %s not found.\n" % fpath
+        )
+        return
+    copyfile(fpath, join(env.subst("$UPLOAD_PORT"), "%s.%s" % (progname, ext)))
+    print(
+        "Firmware has been successfully uploaded.\n"
+        "(Some boards may require manual hard reset)"
+    )
+env.UploadUF2ToDisk = UploadUF2ToDisk
```

Comment:
Is there a real need to store it in the env? I believe you can specify just `UploadUF2ToDisk` in the action list.

Reply:
When I didn't store the function I got an error, I will try your suggestion.

Reply:
Suggestion works + applied.